File: api/model/xrefs/f-review.py
from dataclasses import dataclass
from database import Database


db = Database.get_instance()


@dataclass
class FReview(db.Model):
    _id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, unique=False, nullable=False)
    bottle_id = db.Column(db.Integer, unique=False, nullable=False)
    review_id = db.Column(db.Integer, unique=False, nullable=False)
    value = db.Column(db.Integer, unique=False, nullable=False)

    def __init__(self, user_id, bottle_id, review_id, value) -> None:
        self.user_id = user_id
        self.bottle_id = bottle_id
        self.review_id = review_id
        self.value = value

    def from_json(fReview, user_id):
        return FReview(
            user_id,
            fReview["bottle_id"],
            fReview["review_id"],
            fReview["value"],
        )

File: api/model/f-review.ts
import { Entity, Field, Nullable, PrimaryKey } from "../../deps.ts";

@Entity("f_review")
export class FReview {
  constructor(
    @PrimaryKey("SERIAL") public _id: number,
    @Field("INT", Nullable.NO, "account_id") public accountId: number,
    @Field("INT", Nullable.NO, "bottle_id") public bottleId: number,
    @Field("INT", Nullable.NO, "review_id") public reviewId: number,
    @Field("INT") public value: number,
  ) {
  }
}
